documentation/configuration.md

```diff
@@ -49,22 +49,70 @@ kubectl create secret generic apikey \
   | kubectl apply -f -
 ```
 
-You can then use a Kubernetes volume mount to supply the secret to the container at runtime
+You can then use a Kubernetes volume mount to supply the secret to the container at runtime.
 
 ```yaml 
 spec:
   volumes:
-  - name: apikey
+  - name: config
     secret:
       secretName: apikey
   containers:
   - name: dotnetmonitoragent
     image: mcr.microsoft.com/dotnet/dotnet-monitor:5.0.0-preview.4
     volumeMounts:
-      - name: apikey
+      - name: config
+        mountPath: /etc/dotnet-monitor
+```
+
+Alternatively, you can also configuration maps to specify configuration to the container at runtime.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-configmap
+data:
+  Metrics__MetricCount: "2"
```

Comment:
I think we should have a more intuitive configuration here, such as Metrics__UpdateIntervalSeconds. The other option would be to demonstrate how an egress provider can separate its secret and non-secret data.

Reply:
In general the issue with scenario specific docs (like the egress provider) is that they tend to go stale when the config schema changes. As a first step, let's just have good conceptual docs.

I'm happy to reconsider once we've completed all the basics 😄